eventservice/api/utils/group_auth.py

```diff
@@ -26,14 +26,12 @@ def get_group_ids():
 def get_group_memberships():
     group_memberships = list()
     include_private_events = False
-    # user UserAuth request
-    if 'user_token_data' in g:
+    if 'user_token' in g and not g.user_token_data.get('anonymous'):
```

Comment:
Hi, @shurwit, 

The new group membership endpoint is returning 401 with the tokens signed by Rokwire issuer api.rokwire.illinois.edu. As you know, this was the previous endpoint - /gr/api/int/user/{UIN}/groups. When UIN is None (since there is no logged-in user here), the previous endpoint would return an empty list along with success code 200. Should we continue to use that endpoint while the service tokens feature is being added? Or do you have other suggestions? This is also applicable to get_group_ids method above. Please let us know. Thanks.

CC: @ywkim312

Reply:
Hi @sandeep-ps @ywkim312, yes you are correct that the new endpoint will only accept core tokens. I think our best bet for now is to determine if the token is a Core token here. If it is, we will use the new endpoint as it does now. If not, it will use the old endpoint. To do this we could check the issuer of the token. Let me know if you see any issues with this approach or if you have any questions. Thanks!

Reply:
Okay. Thanks. Yes, it makes sense to check for the issuer and make different calls based on that.

@ywkim312, let's have a chat to get this clarified.

Reply:
Sorry @sandeep-ps, I realized I made a mistake in my last post. The new endpoint will accept either Core or Shibboleth tokens. It is important that we continue sending Core tokens to this endpoint to support non-Shibboleth authenticated users. I do not think it really matters which endpoint we use for Shibboleth tokens, but I would suggest leaving them on the new endpoint just to be safe. The Rokwire issuer tokens will need to go to the old endpoint as I described. Hopefully this helps!

Reply:
Yes, this really helps. Thank you!